### ISSUE TITLE
Social Web + Browsers = <3

### ISSUE THREAD

**Issue by samuelgoto** [w3c-fedid/idp-registration#15]
> These are some personal notes I took from a session at IIW and a few exploratory designs that occurred to me while meeting folks there.

@aaronpk and I hosted a session at IIW on the topic of ways in which the browser could help the the fediverse succeed.

We discussed a few things, like being able to being able to facilitate the discovery of your ActivityPub instance when you are interacting with another ActivtyPub instance.

For example, we discussed how we all had trouble with the unnecessary friction involved in discovering your instance by manually typing it:

<img width="894" alt="Screenshot 2024-11-01 at 11 54 59" src="https://github.com/user-attachments/assets/4663e3f7-572f-413a-af26-ff2a81807fa2">

@aaronpk and mine's intuition was that this problem is systemic across every federated system, and equivalent to a proposal on how to make this better for the Indie Web:

<img width="903" alt="Screenshot 2024-11-01 at 11 57 23" src="https://github.com/user-attachments/assets/03dc4d0a-a29a-42bd-9ec6-f46dde8c7c0e">

How the browser materializes this UX is obviously its own choice, but I think there is something powerful that happens when we transpose this to the URL bar:

![Wrapper-9](https://github.com/user-attachments/assets/1124f0db-f2b5-436b-8eb6-474eb49fffc9)

I think once you get things into the URL bar, it can become an anchor to a lot more social gestures between websites. 

For example:

![Wrapper-8](https://github.com/user-attachments/assets/b5f96ecc-955e-413c-aef7-607b3cf3ece5)
![Wrapper-11](https://github.com/user-attachments/assets/86e3cd2b-c2b0-4634-9ed0-ea034dac6d98)
![Wrapper-12](https://github.com/user-attachments/assets/4b7c0cf1-2e16-48bb-831c-5e98ef2d7104)
![Wrapper-17](https://github.com/user-attachments/assets/5b1ecbf6-81c7-4ca0-bff0-7c0314c2df39)
![Wrapper-18](https://github.com/user-attachments/assets/cab138cc-2c8a-4004-8b7e-959897d20106)
![Wrapper-19](https://github.com/user-attachments/assets/ddcf5ad9-0066-48af-a380-127d4d27343a)

This translates well too for instances you don't operate yourself:

![Wrapper-13](https://github.com/user-attachments/assets/b8e45543-4549-4acd-b97b-b6d441dfb929)
![Wrapper-14](https://github.com/user-attachments/assets/5b0a426a-c670-45c8-aa1b-3dad7930f2dd)
![Wrapper-15](https://github.com/user-attachments/assets/c4553d1b-5f52-44bd-a439-fc940392ea8a)
![Wrapper-16](https://github.com/user-attachments/assets/69852e17-dd8f-47cc-a263-5c72b0ede288)

Or even actions that aren't even that social:

![Wrapper-21](https://github.com/user-attachments/assets/ec41772f-a3b8-4436-b66b-b5e5b2f6b121)
![Wrapper-22](https://github.com/user-attachments/assets/563e1ca4-728b-4a0f-9abd-384c543075e0)
![Wrapper-23](https://github.com/user-attachments/assets/33a70e67-ad6a-4491-a843-45f8a598968f)

There are a lot of open UX design questions here, here are a few formulations to give you a sense of other materializations:

![Wrapper-1](https://github.com/user-attachments/assets/fd55bda0-293d-4148-9a70-76e0b97b5fd0)
![Wrapper-2](https://github.com/user-attachments/assets/a37cd11a-d80a-45fb-9849-3c952669289a)
![Wrapper-3](https://github.com/user-attachments/assets/8226166f-6028-498b-8d1b-bf713b944341)
![Wrapper-4](https://github.com/user-attachments/assets/b33948f5-c0cf-4be9-a7a2-f21a03620894)
![Wrapper-5](https://github.com/user-attachments/assets/683ce6c7-4c22-4dcf-afc8-f93530c3b591)
![Wrapper-6](https://github.com/user-attachments/assets/c0675d9a-7ad1-4b18-b7c3-5311558a714d)
![Wrapper-7](https://github.com/user-attachments/assets/12d0d87b-0e19-437c-8ded-0e65f1b95044)
![Wrapper-10](https://github.com/user-attachments/assets/8713e24f-fcb6-4b1c-8700-7740bd468413)
![Wrapper-20](https://github.com/user-attachments/assets/e16a7589-3ad1-48aa-9a4c-c9b8801eeba9)
![Wrapper](https://github.com/user-attachments/assets/f45ac0f1-2dfc-4bd2-986f-8ce281eba5ae)

Regardless of which specific materialization we pick, I just wanted to write down an idea that occurred during IIW that I think could be pretty cool to explore further.

At this point, this is just an early idea that I wanted to get out there to get a sense of what folks may think.

WDYT?

**Comment by elf-pavlik** [comment]
I like the general idea! It feels like hypermedia with affordances.
It also reminds me of a TPAC session back in 2014 when there was an attempt to reconcile work in Social WG, especially ActivityStreams 2.0 and https://schema.org/Action

https://www.w3.org/2014/10/29-social-minutes.html

**Comment by bumblefudge** [comment]
![Screenshot_20241104_190757_GitHub.jpg](https://github.com/user-attachments/assets/d30fbc04-c53a-4abf-8288-9d4a7777ef0b)

but what if you're logged into all 5-- is there a chooser? a list in your settings ranked by generic preference?

**Comment by samuelgoto** [comment]
> but what if you're logged into all 5 (these 4 plus your own domain auth system)-- is there a chooser? a list in your settings ranked by generic preference? a favicon.ico pull down?

Good question, I dunno :) I think it is equivalent to some UX problems we are solving with https://github.com/w3c-fedid/multi-idp, so seems solvable to me. But yeah, agreed, hard UX problem here.

**Comment by samuelgoto** [comment]
> It also reminds me of a TPAC session back in 2014 when there was an attempt to reconcile work in Social WG, especially ActivityStreams 2.0 and https://schema.org/Action

Ha, you found my 15 year long plan!! :) You can't step on the same river twice, so maybe we'll manage to affect change now!

> https://schema.org/docs/about.html
> Sam Goto of Google did much of the work being schema.org's Actions vocabulary.

**Comment by chrismessina** [comment]
This approach wouldn't work well in browsers that hide or deprioritize the URL bar, like Arc.

**Comment by samuelgoto** [comment]
> This approach wouldn't work well in browsers that hide or deprioritize the URL bar, like Arc.

Yeah, each browser would have to figure out ways to anchor these. It seems like a solvable UX problem though, doesn't it @chrismessina ?

On a related note, good running into you online after so many years (more than 10?) @chrismessina !

**Comment by elf-pavlik** [comment]
I'm wondering if this could be built on top of [Web Share API](https://www.w3.org/TR/web-share/) and [Web Share Target API](https://w3c.github.io/web-share-target/) to some extent by adding the semantics of different types of actions.

**Comment by thhck** [comment]
exiting idea !

> ![Image](https://github.com/user-attachments/assets/f5ed59ff-bc1c-4e33-8745-f9865ee98966)

Feels like bringing RSS back ! I miss RSS ... :)

**Comment by samuelgoto** [comment]
> Feels like bringing RSS back ! I miss RSS ... :)

i miss RSS too :)

**Comment by thhck** [comment]
I think decentralised ecosystem can be confusing for user, and having actions like "follow," "like," and "sign in" unified through the browser's UI could indeed simplify interactions and improve UX for sure.

However I'm wondering what would be the cost of delegating these UI elements to the browser. For example ( assuming a world where such a system is implemented and users are accustomed to this browser interface ) introducing a new "social web" protocol would require convincing all browser vendors to incorporate these actions with the new protocol into their UI. Or to implement them as regular html button, but with a feeling of being sort of second class interaction.

**Comment by andypiper** [comment]
I have an observation here that the [Graze plugin for Chrome](https://chromewebstore.google.com/detail/graze/epocinhmkcnjfjobnglchpbncndobblj) does something similar to the UI concepts shown above and provides discovery and 1-click follow, so this is certainly something that can be finessed through a browser helper. Whether that is the "right" place to do it is another matter, but I like the idea of discovering the additional identities and offering actions for them in the browser.
